src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java

```diff
@@ -21,76 +26,168 @@
 
     private static final IMovementStrategy MOVE_UP = new UpMovementStrategy();
     private static final IMovementStrategy STATIONARY = new StationaryMovementStrategy();
-
-    private static final int BOARD_DIMENSION_EQUALS_TWO = 2;
-    private static final int MAX_ROUNDS_EQUALS_TWO = 2;
-    private static final int NUMBER_OF_BEACONS_EQUALS_ONE = 1;
-    private static final int NUMBER_OF_OBSERVERS_EQUALS_ONE = 1;
     private static final double TRANSMISSION_THRESHOLD_RADIUS_EQUALS_ONE = 1.0;
     private static final int AWAKENESS_CYCLE_EQUALS_TWO = 2;
     private static final int AWAKENESS_DURATION_EQUALS_ONE = 1;
-    private static final AwakenessStrategyFactory.Type FIXES_AWAKENESS_STRATEGY_TYPE =
-            AwakenessStrategyFactory.Type.FIXED;
+    private static final AwakenessStrategyFactory.Type FIXED_AWAKENESS_STRATEGY_TYPE = AwakenessStrategyFactory.Type.FIXED;
 
     @Before
     public void setUp() {
         helper.setUp();
     }
 
     @Test
-    public void runSimulationSingleRoundVerifyAgentsLocations() {
-        // Create new simulation.
+    public void runTwoRoundsSimulationVerifyMovingUpAgentsLocations() {
+        int roundsNum = 2;
+        int rowsNum = 5;
+        int colsNum = 5;
+        int beaconsNum = 10;
+        int observersNum = 10;
+
         AbstractSimulation simulation = new TracingSimulation.Builder()
-                .setMaxNumberOfRounds(MAX_ROUNDS_EQUALS_TWO)
-                .setRowNum(BOARD_DIMENSION_EQUALS_TWO)
-                .setColNum(BOARD_DIMENSION_EQUALS_TWO)
-                .setBeaconsNum(NUMBER_OF_BEACONS_EQUALS_ONE)
-                .setObserversNum(NUMBER_OF_OBSERVERS_EQUALS_ONE)
+                .setMaxNumberOfRounds(roundsNum + 1) // The first round is the initialization
+                .setRowNum(rowsNum)
+                .setColNum(colsNum)
+                .setBeaconsNum(beaconsNum)
+                .setObserversNum(observersNum)
                 .setTransmissionThresholdRadius(TRANSMISSION_THRESHOLD_RADIUS_EQUALS_ONE)
                 .setBeaconMovementStrategy(MOVE_UP)
+                .setObserverMovementStrategy(MOVE_UP)
+                .setAwakenessCycle(AWAKENESS_CYCLE_EQUALS_TWO)
+                .setAwakenessDuration(AWAKENESS_DURATION_EQUALS_ONE)
+                .setAwakenessStrategyType(FIXED_AWAKENESS_STRATEGY_TYPE)
+                .build();
+
+        Map<String, Location> initialAgentsToLocations = mapAgentsToLocationsOnBoard(simulation.getRealBoardState());
+
+        simulation.run();
+        Map<String, Location> terminalAgentsToLocations = mapAgentsToLocationsOnBoard(simulation.getRealBoardState());
+
+        for (String agent : initialAgentsToLocations.keySet()) {
+            assertThat(terminalAgentsToLocations.get(agent)).isEqualTo(
```

Comment:
Please verify the line-breaks / indentation here - I believe the recommended style would move the "`.isEqualTo(`" to the next line (and combine it with the value there).